ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Util/AnchoredFragment.hs

```diff
@@ -79,10 +81,15 @@ forksAtMostKBlocks k ours theirs = case ours `AF.intersect` theirs of
 -- these fragments intersect with our current chain, they must by transitivity
 -- also intersect each other.
 compareAnchoredFragments ::
-     forall blk. (BlockSupportsProtocol blk, HasCallStack)
+     forall blk t.
+     ( BlockSupportsProtocol blk
+     , HasCallStack
+     , ProjectHeader t blk
+     , HasHeader (t blk)
+     )
   => BlockConfig blk
-  -> AnchoredFragment (Header blk)
-  -> AnchoredFragment (Header blk)
+  -> AnchoredFragment (t blk)
+  -> AnchoredFragment (t blk)
```

Comment:
The payloads could also be different, right?

Reply:
Good one. I think so, let me try.